### PR TITLE
Optimized modularization of `auth` command

### PR DESCRIPTION
**Pull request by 0xjei**
This PR brings a net optimization in terms of quantity and readability of code for the `auth` command and related libs. In addition, unnecessary parts in the `actions` package have been removed (since they are non-core) and the remaining ones generalized. This turns out to be a critical step to make easier to write the unit tests and stubs for e2e tests.

This PR also handles the issue related to the expiration of access token #171.

### CODE REVIEW

File: packages/phase2cli/src/lib/utils.ts [hunk 1]
```diff
@@ -29,22 +27,47 @@ import {
...
+ */
+export const getGithubUserHandle = async (githubToken: string): Promise<any> => {
+    // Ask for user account public information through Github API.
+    const response = await request("GET https://api.github.com/user", {
+        headers: {
+            authorization: `token ${githubToken}`
+        }
+    })
+
+    if (response) return response.data.login
```
Comment (ctrlc03) [inline packages/phase2cli/src/lib/utils.ts:66]: might be worth checking the status code of the response before returning?

File: packages/actions/src/helpers/firebase.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+/**
+ * Return the current authenticated user in the given Firebase Application.
+ * @param firebaseApp <FirebaseApp> - the configured instance of the Firebase App in use.
+ * @returns
+ */
+export const getCurrentFirebaseAuthUser = (firebaseApp: FirebaseApp): User => {
+    const user = getAuth(firebaseApp).currentUser
+
+    if (!user) throw new Error(`Authenticated user not found for current Firebase application`)
```
Comment (ctrlc03) [inline packages/actions/src/helpers/firebase.ts:77]: should we keep using error codes? e.g. Firebase-0001 - Authenticated ...

File: packages/phase2cli/src/lib/utils.ts (hunk 1 above)
Comment (0xjei) [reply packages/phase2cli/src/lib/utils.ts:66]: Correct! Thanks for the heads up, will fix in next commit.

File: packages/actions/src/helpers/firebase.ts (hunk 2 above)
Comment (0xjei) [reply packages/actions/src/helpers/firebase.ts:77]: Not anymore. We are going to make agnostic and method only related errors in the `actions` package. These errors can be catched in the client/frontend and displayed to the user.

File: packages/phase2cli/src/lib/utils.ts (hunk 1 above)
Comment (0xjei) [reply packages/phase2cli/src/lib/utils.ts]: Fixed [bf0fbab](https://github.com/quadratic-funding/mpc-phase2-suite/pull/270/commits/bf0fbaba3ab89b6d57dcd4dfa410129e55e04a2c)

**Review by ctrlc03 — APPROVED** [review]

**Pull request merged**